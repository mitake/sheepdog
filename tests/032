#!/bin/bash

# Test recovery when we have less zones availability

seq=`basename $0`
echo "QA output created by $seq"

here=`pwd`
tmp=/tmp/$$
status=1        # failure is the default!

# get standard environment, filters and checks
. ./common.rc
. ./common.filter

_cleanup

for i in `seq 0 7`; do
    _start_sheep $i
done

_wait_for_sheep 8

$COLLIE cluster format

$COLLIE vdi create test 100MB
# create 25 objects
for i in `seq 0 24`; do
    echo $i | $COLLIE vdi write test $((i * 4 * 1024 * 1024)) 512
done

$COLLIE vdi read test | md5sum

for i in `seq 7 -1 0`; do
    _wait_for_sheep_recovery 0
    _kill_sheep $i
done

sleep 1

# start master first to avoid master transfer
_start_sheep 0
_wait_for_sheep 1
for i in `seq 0 7`; do
    _start_sheep $i
done

_wait_for_sheep_recovery 0

ls $STORE/*/obj/807c2b2500000000 | sort
for i in `seq 0 24`; do
    ls $STORE/*/obj/007c2b25000000`printf "%02x" $i` | sort
done
ls $STORE/*/obj/.stale | sort

$COLLIE vdi read test | md5sum
