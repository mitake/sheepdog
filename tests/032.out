QA output created by 032
using backend farm store
9c7766570b3be3aff2724f587c2f4107  -
/tmp/sheepdog/0/obj/807c2b2500000000
/tmp/sheepdog/1/obj/807c2b2500000000
/tmp/sheepdog/2/obj/807c2b2500000000
/tmp/sheepdog/0/obj/007c2b2500000000
/tmp/sheepdog/1/obj/007c2b2500000000
/tmp/sheepdog/2/obj/007c2b2500000000
/tmp/sheepdog/0/obj/007c2b2500000001
/tmp/sheepdog/1/obj/007c2b2500000001
/tmp/sheepdog/4/obj/007c2b2500000001
/tmp/sheepdog/0/obj/007c2b2500000002
/tmp/sheepdog/1/obj/007c2b2500000002
/tmp/sheepdog/4/obj/007c2b2500000002
/tmp/sheepdog/2/obj/007c2b2500000003
/tmp/sheepdog/6/obj/007c2b2500000003
/tmp/sheepdog/7/obj/007c2b2500000003
/tmp/sheepdog/0/obj/007c2b2500000004
/tmp/sheepdog/3/obj/007c2b2500000004
/tmp/sheepdog/4/obj/007c2b2500000004
/tmp/sheepdog/0/obj/007c2b2500000005
/tmp/sheepdog/3/obj/007c2b2500000005
/tmp/sheepdog/5/obj/007c2b2500000005
/tmp/sheepdog/4/obj/007c2b2500000006
/tmp/sheepdog/5/obj/007c2b2500000006
/tmp/sheepdog/7/obj/007c2b2500000006
/tmp/sheepdog/3/obj/007c2b2500000007
/tmp/sheepdog/5/obj/007c2b2500000007
/tmp/sheepdog/7/obj/007c2b2500000007
/tmp/sheepdog/1/obj/007c2b2500000008
/tmp/sheepdog/3/obj/007c2b2500000008
/tmp/sheepdog/6/obj/007c2b2500000008
/tmp/sheepdog/3/obj/007c2b2500000009
/tmp/sheepdog/5/obj/007c2b2500000009
/tmp/sheepdog/7/obj/007c2b2500000009
/tmp/sheepdog/0/obj/007c2b250000000a
/tmp/sheepdog/2/obj/007c2b250000000a
/tmp/sheepdog/5/obj/007c2b250000000a
/tmp/sheepdog/1/obj/007c2b250000000b
/tmp/sheepdog/4/obj/007c2b250000000b
/tmp/sheepdog/5/obj/007c2b250000000b
/tmp/sheepdog/3/obj/007c2b250000000c
/tmp/sheepdog/4/obj/007c2b250000000c
/tmp/sheepdog/5/obj/007c2b250000000c
/tmp/sheepdog/0/obj/007c2b250000000d
/tmp/sheepdog/2/obj/007c2b250000000d
/tmp/sheepdog/3/obj/007c2b250000000d
/tmp/sheepdog/1/obj/007c2b250000000e
/tmp/sheepdog/3/obj/007c2b250000000e
/tmp/sheepdog/7/obj/007c2b250000000e
/tmp/sheepdog/2/obj/007c2b250000000f
/tmp/sheepdog/4/obj/007c2b250000000f
/tmp/sheepdog/5/obj/007c2b250000000f
/tmp/sheepdog/0/obj/007c2b2500000010
/tmp/sheepdog/1/obj/007c2b2500000010
/tmp/sheepdog/4/obj/007c2b2500000010
/tmp/sheepdog/3/obj/007c2b2500000011
/tmp/sheepdog/5/obj/007c2b2500000011
/tmp/sheepdog/7/obj/007c2b2500000011
/tmp/sheepdog/2/obj/007c2b2500000012
/tmp/sheepdog/4/obj/007c2b2500000012
/tmp/sheepdog/6/obj/007c2b2500000012
/tmp/sheepdog/0/obj/007c2b2500000013
/tmp/sheepdog/5/obj/007c2b2500000013
/tmp/sheepdog/6/obj/007c2b2500000013
/tmp/sheepdog/0/obj/007c2b2500000014
/tmp/sheepdog/5/obj/007c2b2500000014
/tmp/sheepdog/6/obj/007c2b2500000014
/tmp/sheepdog/0/obj/007c2b2500000015
/tmp/sheepdog/2/obj/007c2b2500000015
/tmp/sheepdog/3/obj/007c2b2500000015
/tmp/sheepdog/1/obj/007c2b2500000016
/tmp/sheepdog/5/obj/007c2b2500000016
/tmp/sheepdog/6/obj/007c2b2500000016
/tmp/sheepdog/2/obj/007c2b2500000017
/tmp/sheepdog/5/obj/007c2b2500000017
/tmp/sheepdog/6/obj/007c2b2500000017
/tmp/sheepdog/0/obj/007c2b2500000018
/tmp/sheepdog/1/obj/007c2b2500000018
/tmp/sheepdog/5/obj/007c2b2500000018







/tmp/sheepdog/0/obj/.stale:
/tmp/sheepdog/1/obj/.stale:
/tmp/sheepdog/2/obj/.stale:
/tmp/sheepdog/3/obj/.stale:
/tmp/sheepdog/4/obj/.stale:
/tmp/sheepdog/5/obj/.stale:
/tmp/sheepdog/6/obj/.stale:
/tmp/sheepdog/7/obj/.stale:
9c7766570b3be3aff2724f587c2f4107  -
